### PR TITLE
Fixing pagination ( Go to page functionality)

### LINKED ISSUE #135: Bug Fix - Pagination
1) Pagination is breaking in mobile device 

2) "Go to page:" is not working as expected 
Not able to jump to page 22 or any other page not starting with 1 , not able to remove 1
image

### PR DESCRIPTION
**Pull request by Suhas-Kalagotla**
1. Added node Installation link for windows users in Readme.md  

2. Now able to jump to desire page using "Go to page"
used debounceSearch and it updates after 500ms after user enters input (page number)
image

3. Handling edge cases 
    if user enters values less than 1 
image
it updates to 1 and page 1 is displayed 
image

similarly with input greater than totalPages
image
image

fixes #135

### CODE REVIEW

**Comment by mbashem** [comment]
It runs every 500ms, which is bad for performance. I think there is a better way out there to fix it.  Thank you for your work but sorry I can't approve this.

**Comment by Suhas-Kalagotla** [comment]
In InputNumber component, the debouncedSearch variable will not cause the useDebounce function to run every 500ms . Instead, it will cause debouncedSearch to update with the latest inputValue after 500 milliseconds have passed since the last change to inputValue

**Comment by mbashem** [comment]
Sorry, my bad. But we can get similar result with following code - 
`
  const [inputValue, setInputValue] = useState<string>(props.value.toString());

  const validateAndUpdate = () => {
    let num: number = parseInt(inputValue);
    if (!isNaN(num)) {
      num = processNumber(num, props.min, props.max);
      setInputValue(num.toString());
      props.onChange(num);
    }
  };

  const firstTime = useRef(true);
  useEffect(() => {
    if (firstTime.current) firstTime.current = false;
    else validateAndUpdate();
  }, [inputValue]);
`
It uses useRef hooks to prevent flickering, which you now face in the Go to page box with your implementation.

**Comment by Suhas-Kalagotla** [comment]
made the suggested change 
now using useRef

File: src/util/util.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import { useEffect, useState } from "react";
+
```
Comment (mbashem) [inline src/util/util.ts:2]:
```suggestion
```
This is not necessary anymore.

File: src/util/util.ts [hunk 2]
```diff
@@ -65,7 +67,7 @@ export const delay = (ms) => new Promise((res) => setTimeout(res, ms));
 export const processNumber = (
   num: number,
   min: number,
-  max: number
+  max: number,
```
Comment (mbashem) [inline src/util/util.ts:70]:
```suggestion
  max: number
```
Dont add comma here.

File: src/util/Components/Forms/InputNumber.tsx [hunk 3]
```diff
@@ -29,13 +47,14 @@ const InputNumber = (props: PropsType) => {
...
         placeholder="Max Rating"
-        value={props.value}
+        value={inputValue}
         name={props.name}
         step={props.step ? props.step : 1}
         onChange={(e) => {
+          setInputValue(e.target.value);
           let num: number = parseInt(e.target.value);
 
           num = processNumber(num, props.min, props.max);
```
Comment (mbashem) [inline src/util/Components/Forms/InputNumber.tsx:60]:
```suggestion
        onChange={(e) =>  setInputValue(e.target.value) }
```
Remaining codes are redundant. Rest of the things looks good to me. Nice work!

**Comment by Suhas-Kalagotla** [comment]
made the changes

thank you for your review

File: src/util/Components/Forms/InputNumber.tsx [hunk 4]
```diff
@@ -29,18 +47,14 @@ const InputNumber = (props: PropsType) => {
...
       <input
-        className={"form-control " + props.inputClass}
+        className={"form-control " + (props.inputClass || "")}
         type="number"
         placeholder="Max Rating"
-        value={props.value}
+        value={inputValue}
         name={props.name}
         step={props.step ? props.step : 1}
         onChange={(e) => {
```
Comment (mbashem) [inline src/util/Components/Forms/InputNumber.tsx:56]:
```suggestion
        onChange={(e) => setInputValue(e.target.value)}
```

File: src/util/Components/Forms/InputNumber.tsx [hunk 5]
```diff
@@ -29,18 +47,14 @@ const InputNumber = (props: PropsType) => {
...
-          let num: number = parseInt(e.target.value);
-
-          num = processNumber(num, props.min, props.max);
-
-          props.onChange(num);
+          setInputValue(e.target.value);
         }}
```
Comment (mbashem) [inline src/util/Components/Forms/InputNumber.tsx:58]:
```suggestion
```

File: src/util/Components/Forms/InputNumber.tsx [hunk 6]
```diff
@@ -29,19 +47,14 @@ const InputNumber = (props: PropsType) => {
...
         step={props.step ? props.step : 1}
-        onChange={(e) => {
-          let num: number = parseInt(e.target.value);
-
-          num = processNumber(num, props.min, props.max);
-
-          props.onChange(num);
-        }}
+        onChange={(e) => setInputValue(e.target.value)}
+          setInputValue(e.target.value);
```
Comment (mbashem) [inline src/util/Components/Forms/InputNumber.tsx:57]:
```suggestion
```

**Review by mbashem — APPROVED** [review]
Nice work👏. Thanks for your contribution.

**Pull request merged**